src/Estimators/src/LeggedOdometry.cpp

```diff
@@ -0,0 +1,808 @@
+                return false;
+            }
+
+            A.block(m_spatialDim*cIdx, m_baseOffset, m_spatialDim, m) = m_contactJacobian;
+            int vec3Offset{3};
+            W.block(m_spatialDim*cIdx, m_spatialDim*cIdx, vec3Offset, vec3Offset) *= wLin;
+            W.block(m_spatialDim*cIdx+vec3Offset, m_spatialDim*cIdx+vec3Offset, vec3Offset, vec3Offset) *= wAng;
+            cIdx ++;
+        }
+    }
+
+
+    A.bottomRows(nrJoints) << Eigen::MatrixXd::Zero( nrJoints, m_spatialDim), Eigen::MatrixXd::Identity(nrJoints, nrJoints);
+    W.bottomRightCorner(nrJoints, nrJoints) *= wJVel;
+
+    // least square solution
+    Eigen::MatrixXd S = (A.transpose()*W*A + Delta).inverse();
+    Eigen::VectorXd xOptimal = S*A.transpose()*W*y;
+
+    m_vBase = xOptimal.head(m_spatialDim);
+    m_sDotFilt = xOptimal.tail(nrJoints);
+
+    iDynTree::Vector3 vLin(iDynTree::make_span(m_vBase.head<3>().data(), m_vBase.head<3>().size()));
+    iDynTree::Vector3 vAng(iDynTree::make_span(m_vBase.tail<3>().data(), m_vBase.tail<3>().size()));
+    out.baseTwist.setLinearVec3(vLin);
+    out.baseTwist.setAngularVec3(vAng);
+    return true;
+}
+
+
+bool LeggedOdometry::Impl::computeBaseVelocityUsingAllFixedFrameAverage(const FloatingBaseEstimators::InternalState& state,
+                                                                        FloatingBaseEstimators::Measurements& meas,
+                                                                        FloatingBaseEstimator::ModelComputations& modelComp,
+                                                                        FloatingBaseEstimators::Output& out)
+{
+    const std::string_view printPrefix = "[LeggedOdometry::Impl::computeBaseIMUVelocityUsingAllFixedFrameConstraint] ";
+
+    // here we do the computations with respect to the base link frame and not the base link IMU frame
+    // accordingly, since `m_useIMUVelForBaseVelComputation` is set to false, the IMU velocity in the internal state
+    // remains un-updated, we directly update the base twist in the output
+
+    std::size_t nrContacts{0};
+    // run two loops (alteratively once can do conservative resize of the matrices)
+    Eigen::VectorXd sumV = Eigen::MatrixXd::Zero(6, 1);
+    for (auto& [idx, contact] : state.supportFrameData)
+    {
+        if (contact.isActive)
+        {
+            nrContacts++;
+            if (!modelComp.kinDyn()->getFrameFreeFloatingJacobian(idx, m_contactJacobian))
+            {
+                std::cerr << printPrefix << "Could not compute contact Jacobian."
+                << std::endl;
+                return false;
+            }
+
+            m_contactJacobianBase = m_contactJacobian.block<6, 6>(m_baseOffset, m_baseOffset);
+            m_contactJacobianShape = m_contactJacobian.block(m_baseOffset, m_shapeOffset, m_spatialDim, modelComp.kinDyn()->getNrOfDegreesOfFreedom());
+
+            sumV = -(m_contactJacobianBase.inverse()) * m_contactJacobianShape * meas.encodersSpeed;
```

Comment:
```suggestion
            sumV += -(m_contactJacobianBase.inverse()) * m_contactJacobianShape * meas.encodersSpeed;
```

Reply:
Fixed in https://github.com/dic-iit/bipedal-locomotion-framework/pull/151/commits/d30af6691521cf2e1ef9384582487b16b1dafcef